isalt/scripts.py

```diff
@@ -144,50 +163,69 @@ def main():
         role = 'minion'
     elif args.master:
         role = 'master'
+    elif args.proxy:
+        role = 'proxy'
     on_minion = args.on_minion or bool(os.environ.get(
         'ISALT_ON_MINION',
         isalt_cfg.get('on_minion', 'True')
     ))
     minion_id = None
-    if role == 'minion':
-        cfg_file = args.minion_cfg_file or os.environ.get(
-            'ISALT_MINION_CONFIG',
-            isalt_cfg.get(
-                'minion_config',
-                salt.config.DEFAULT_MINION_OPTS['conf_file'],
-            )
+    master_cfg_file = args.master_cfg_file or os.environ.get(
+        'ISALT_MASTER_CONFIG',
+        isalt_cfg.get(
+            'master_config',
+            salt.config.DEFAULT_MASTER_OPTS['conf_file'],
         )
-        __opts__ = salt.config.minion_config(cfg_file)
+    )
+    master_opts = salt.config.master_config(master_cfg_file)
+    if role in ('minion', 'proxy'):
+        if role == 'minion':
+            cfg_file = args.minion_cfg_file or os.environ.get(
+                'ISALT_MINION_CONFIG',
+                isalt_cfg.get(
+                    'minion_config',
+                    salt.config.DEFAULT_MINION_OPTS['conf_file'],
+                )
+            )
+            __opts__ = salt.config.minion_config(cfg_file)
+        else:
+            cfg_file = args.proxy_cfg_file or os.environ.get(
+                'ISALT_PROXY_MINION_CONFIG',
+                isalt_cfg.get(
+                    'proxy_minion_config',
+                    salt.config.DEFAULT_PROXY_MINION_OPTS['conf_file'],
+                )
+            )
+            __opts__ = salt.config.proxy_config(cfg_file)
         minion_id = args.minion_id or __opts__.get('id',
             os.environ.get('ISALT_MINION_ID', isalt_cfg.get('minion_id'))
         )
+        local = args.local or isalt_cfg.get('local', False)
+        if local and __opts__.get('file_client') != 'local':
+            __opts__['file_client'] = 'local'
+            for opt, value in master_opts.items():
+                if opt.startswith('gitfs_', 'git_', 'azurefs_', 'hgfs_', 'minionfs_', 's3fs_', 'svnfs_', 'pillar_roots', 'file_roots'):
```

Comment:
```
TypeError: startswith() takes at most 3 arguments (9 given)
```

Could use:

```
any([opt.startswith(elem) for elem in ['gitfs_', 'git_', 'azurefs_', 'hgfs_', 'minionfs_', 's3fs_', 'svnfs_', 'pillar_roots', 'file_roots']])
```

Reply:
Ah, good catch... silly me, that should have been a tuple (as a first arg to `startswith`) :man_facepalming:

Reply:
I didn't know you could do that with `startswith`. Nice.